File: InMotionGIT.Common/Helpers/Language.vb
Namespace Helpers

    Public NotInheritable Class Language

        Public Shared Function EnumLanguageToString(full As Boolean) As String
            Dim languagesList As Dictionary(Of Integer, String) = Helpers.Caching.GetItem("Languages")
            Dim result As String = String.Empty

            For Each languageItem As KeyValuePair(Of Integer, String) In languagesList
                If Not String.IsNullOrEmpty(result) Then
                    result += ","
                End If

                result += languageItem.Key.ToString
            Next

            Return result
        End Function

        'Public Shared Function EnumLanguageToDictionary(full As Boolean) As Dictionary(Of Integer, String)
        '    Return EnumLanguageToDictionary(EnumLanguage.English, full)
        'End Function

        Public Shared Function CultureNamesToEnumLanguage(codeLanguage As String) As Integer
            Dim result As Integer

            If codeLanguage.ToLower.StartsWith("es") Then
                result = 2

            ElseIf codeLanguage.ToLower.StartsWith("en") Then
                result = 1

            ElseIf codeLanguage.ToLower.StartsWith("pt") Then
                result = 3
            Else
                result = 0
            End If

            Return result
        End Function

        Public Shared Function DescriptionToEnumLanguage(description As String) As Integer

            Return DescriptionToEnumLanguage(description, 1)
        End Function

        Public Shared Function DescriptionToEnumLanguage(description As String, language As Integer) As Integer
            Dim result As Integer
            Dim languagesList As Dictionary(Of Integer, String) = Helpers.Caching.GetItem("Languages")

            For Each languageItem As KeyValuePair(Of Integer, String) In languagesList
                If String.Equals(languageItem.Value, description, StringComparison.CurrentCultureIgnoreCase) Then

                    result = languageItem.Key

                    Exit For
                End If
            Next

            Return result
        End Function

        ''' <summary>
        ''' Extracts the name of the enum Cultural Language/Extrae el cultural name del enum de lenguaje
        ''' </summary>
        ''' <param name="language">Lenguaje/Language</param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Shared Function LanguageToCultureName(language As Integer) As String
            Dim result As String = String.Empty
            Select Case language
                Case 1
                    result = "EN-US"

                Case 2
                    result = "ES-CR"

                Case Else
                    result = "PT-BR"
            End Select

            Return result
        End Function

        ''' <summary>
        '''
        ''' </summary>
        ''' <param name="CultureName"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Shared Function CultureNameToLanguage(CultureName As String) As Integer
            Dim result As Integer

            If CultureName.StartsWith("en", StringComparison.CurrentCultureIgnoreCase) Then
                result = 1
            ElseIf CultureName.StartsWith("es", StringComparison.CurrentCultureIgnoreCase) Then
                result = 2
            Else
                result = 3
            End If

            Return result
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/Helpers/Language.cs
using System;
using System.Collections.Generic;

namespace InMotionGIT.Common.Helpers
{

    public sealed class Language
    {

        public static string EnumLanguageToString(bool full)
        {
            Dictionary<int, string> languagesList = (Dictionary<int, string>)Caching.GetItem("Languages");
            string result = string.Empty;

            foreach (KeyValuePair<int, string> languageItem in languagesList)
            {
                if (!string.IsNullOrEmpty(result))
                {
                    result += ",";
                }

                result += languageItem.Key.ToString();
            }

            return result;
        }

        // Public Shared Function EnumLanguageToDictionary(full As Boolean) As Dictionary(Of Integer, String)
        // Return EnumLanguageToDictionary(EnumLanguage.English, full)
        // End Function

        public static int CultureNamesToEnumLanguage(string codeLanguage)
        {
            int result;

            if (codeLanguage.ToLower().StartsWith("es"))
            {
                result = 2;
            }

            else if (codeLanguage.ToLower().StartsWith("en"))
            {
                result = 1;
            }

            else if (codeLanguage.ToLower().StartsWith("pt"))
            {
                result = 3;
            }
            else
            {
                result = 0;
            }

            return result;
        }

        public static int DescriptionToEnumLanguage(string description)
        {

            return DescriptionToEnumLanguage(description, 1);
        }

        public static int DescriptionToEnumLanguage(string description, int language)
        {
            var result = default(int);
            Dictionary<int, string> languagesList = (Dictionary<int, string>)Caching.GetItem("Languages");

            foreach (KeyValuePair<int, string> languageItem in languagesList)
            {
                if (string.Equals(languageItem.Value, description, StringComparison.CurrentCultureIgnoreCase))
                {

                    result = languageItem.Key;

                    break;
                }
            }

            return result;
        }

        /// <summary>
        /// Extracts the name of the enum Cultural Language/Extrae el cultural name del enum de lenguaje
        /// </summary>
        /// <param name="language">Lenguaje/Language</param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static string LanguageToCultureName(int language)
        {
            string result = string.Empty;
            switch (language)
            {
                case 1:
                    {
                        result = "EN-US";
                        break;
                    }

                case 2:
                    {
                        result = "ES-CR";
                        break;
                    }

                default:
                    {
                        result = "PT-BR";
                        break;
                    }
            }

            return result;
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="CultureName"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static int CultureNameToLanguage(string CultureName)
        {
            int result;

            if (CultureName.StartsWith("en", StringComparison.CurrentCultureIgnoreCase))
            {
                result = 1;
            }
            else if (CultureName.StartsWith("es", StringComparison.CurrentCultureIgnoreCase))
            {
                result = 2;
            }
            else
            {
                result = 3;
            }

            return result;
        }

    }

}